core/src/main/kotlin/org/evomaster/core/problem/external/service/ExternalServiceHandler.kt

```diff
@@ -0,0 +1,168 @@
+     * generate a new one.
+     */
+    private fun generateRandomAvailableAddress(port: Int) : String {
+        val ip = generateRandomIPAddress(randomness)
+        if (isAddressAvailable(ip, port)) {
+            return ip
+        }
+        return generateRandomAvailableAddress(port)
+    }
+
+    fun getExternalServices() : List<ExternalServiceInfo> {
+        return externalServiceInfos
+    }
+
+    /**
+     * Default IP address will be a randomly generated IP
+     *
+     * If user provided IP address isn't available on the port
+     * IllegalStateException will be thrown.
+     */
+    private fun getIP(port: Int) : String {
+        val ip: String
+        when (config.externalServiceIPSelectionStrategy) {
+            // Although the default address will be a random, this
+            // option allows selecting explicitly
+            EMConfig.ExternalServiceIPSelectionStrategy.RANDOM -> {
+                ip = if (externalServiceInfos.size > 0) {
+                    getNextAvailableAddress(port)
+                } else {
+                    generateRandomAvailableAddress(port)
+                }
+            }
+            EMConfig.ExternalServiceIPSelectionStrategy.USER -> {
+                ip = if (externalServiceInfos.size > 0) {
+                    getNextAvailableAddress(port)
+                } else {
+                    if (!isReservedIP(config.externalServiceIP)) {
+                        if (isAddressAvailable(config.externalServiceIP, port)) {
+                            config.externalServiceIP
+                        } else {
+                            throw IllegalStateException("User provided IP address is not available")
+                        }
+                    } else {
+                        throw IllegalStateException("Can not use a reserved IP address")
+                    }
+                }
+            }
+            else -> {
+                ip = if (externalServiceInfos.size > 0) {
+                    getNextAvailableAddress(port)
+                } else {
+                    generateRandomAvailableAddress(port)
+                }
+            }
+        }
+        return ip
+    }
+    /**
+     * Will initialise WireMock instance on a given IP address for a given port.
+     */
+    private fun initWireMockServer(address: String, port: Int): WireMockServer {
+        // TODO: Port need to be changed to the remote service port
+        // In CI also using remote ports as 80 and 443 fails
+        val wm = WireMockServer(
+            WireMockConfiguration()
+                .bindAddress(address)
+                .port(port)
+                .extensions(ResponseTemplateTransformer(false)))
+        wm.start()
+
+        // to prevent from the 404 when no matching stub below stub is added
+        wm.stubFor(get(urlMatching("/.*"))
+            .atPriority(2)
+            .willReturn(aResponse()
+                .withStatus(200)
```

Comment:
hmmm... leave a TODO. we ll need to discss what should be the default behavior when WM not configured yet. eg, maybe 500 or 404 could be better than 200. we can discuss it later on (ie, for now just leave a TODO)